integrations/jenkins/main.py

```diff
@@ -57,6 +58,15 @@ async def on_resync_users(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
         yield users
 
 
+@ocean.on_resync(ObjectKind.STAGE)
+async def on_resync_stages(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    jenkins_client = init_client()
+
+    async for stages in jenkins_client.get_stages():
+        logger.info(f"Received batch with {len(stages)} stages")
+        yield stages
+
```

Comment:
One of the comment we received was that we should ensure we don't ingest million records into Port. Can we add the check here so that we cap the stages entity to 10k builds?